### ISSUE TITLE
Flexible title (and other) attributes for dbc.Offcanvas elements

### ISSUE THREAD

**Issue by wbeardall** [facultyai/dash-bootstrap-components#936]
Currently, the implementation of the dbc.Offcanvas title attribute only accepts a string, and there appears to be no way to modify the styling of this part of the component. This significantly limits the user's options when creating aesthetic interfaces, especially when using styled contents in the offcanvas. For example, using boldened and enlarged subheaders within the Offcanvas body makes the header look very out of place, and naturally there is no way to ensure consistent fonts or text styles between the two. A visual example of this problem is provided below.

![Style mismatch between content and title of Offcanvas](https://user-images.githubusercontent.com/37509266/218617378-6eec1448-e2cb-4b52-a44d-bd8c061c249e.png)

The lack of ability to override styles for the title (and header generally) also causes clashes with external_scripts. For example, when utilising Tailwind for classes in an app (as below), the close button in the dbc.Offcanvas becomes invisible, and there is no way to override the style to fix it. (See above, the component's close button is invisible, although it is still usable).

```
app = Dash(__name__,external_scripts=["https://cdn.tailwindcss.com"])
```

It would be very useful to be able to specify an HTML element for the title attribute of the dbc.Offcanvas, and to be able to override the styles of both the header and individual subcomponents.

**Comment by tcbegley** [comment]
Hi @wbeardall 

I just made a pre-release that makes it possible to use arbitrary components as the `title` prop. Please do try it out and let me know how you get one

```
pip install dash-bootstrap-components==1.3.2rc1
```

**Comment by wbeardall** [comment]
Thanks a bunch, that works just as expected. For anyone else encountering this, the behaviour is fixed in main release version 1.4.0.
